yjit/src/core.rs

```diff
@@ -339,11 +339,53 @@ type BranchGenFn =
 
 /// A place that a branch could jump to
 #[derive(Debug)]
-struct BranchTarget {
+enum BranchTarget {
+    Stub(Box<BranchStub>), // Not compiled yet
+    Block(BlockRef),       // Already compiled
+}
+
+impl BranchTarget {
+    fn get_address(&self) -> Option<CodePtr> {
+        match self {
+            BranchTarget::Stub(stub) => stub.address,
+            BranchTarget::Block(blockref) => blockref.borrow().start_addr,
+        }
+    }
+
+    fn get_id(&self) -> BlockId {
+        match self {
+            BranchTarget::Stub(stub) => stub.id,
+            BranchTarget::Block(blockref) => blockref.borrow().blockid,
+        }
```

Comment:
I think I'd prefer if we call this `get_blockid()` just for consistency with the rest of the codebase.

Everything else looks good 👍

Reply:
:+1: https://github.com/ruby/ruby/pull/7151/commits/a9f0bb589127cd58ac0ffb455fd6a24354ca71f9